File: SoloLearn Codes/Missing Numbers/MissingNumbers.java
import java.util.*;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class MissingNumbers
{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        List<Integer> numberSequence = Arrays
                .stream(new int[n])
                .map(a -> sc.nextInt())
                .boxed()
                .toList();
        System.out.println(IntStream
                .range(numberSequence
                        .stream()
                        .min(Comparator.comparingInt(a -> a))
                        .orElse(0), numberSequence.stream()
                        .max(Comparator.comparingInt(a -> a))
                        .orElse(0))
                .filter(a -> !numberSequence.contains(a))
                .mapToObj(String::valueOf)
                .collect(Collectors.joining(" ")));
	}
}

File: SoloLearn Codes/Missing Numbers/missing_numbers.swift
import Foundation

let n = Int(readLine()!)!
var nums = (1...n)
    .map{ x in Int(readLine()!)! }
    .sorted()
print((nums.min()!...nums.max()!)
    .filter{ !nums.contains($0) }
    .map{ String($0) }
    .joined(separator: " "))